src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -108,29 +109,14 @@ public new void RemoveMetadataReference(string referencePath)
         public void AddProjectReference(IWorkspaceProjectContext project, MetadataReferenceProperties properties)
         {
             var abstractProject = GetAbstractProject(project);
-
-            // AbstractProject and ProjectTracker track project references using the project bin output path.
-            // Setting the command line arguments should have already set the output file name and folder.
-            // We fetch this output path to add the reference.
-            var referencedProject = this.ProjectTracker.GetProject(abstractProject.Id);
-            var binPathOpt = referencedProject.BinOutputPath;
-            if (!string.IsNullOrEmpty(binPathOpt))
-            {
-                AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible(binPathOpt, properties);
-            }
+            AddProjectReference(new ProjectReference(abstractProject.Id, properties.Aliases, properties.EmbedInteropTypes));
```

Comment:
This was my original implementation, but AddProjectReference didn't make updates to `_metadataFileNameToConvertedProjectReference`, while AddMetadataReferenceAndTryConvertingToProjectReferenceIfPossible took care of it, see [here](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/AbstractProject.cs,545).